### DISCUSSION TITLE
CastError: Cast to ObjectId failed for value "undefined" (type string) at path "_id" for model "Comments"

### DISCUSSION THREAD

**Discussion by jonatsuji** [neuefische/questions#455] [Web] [answered]
I get the error message you see above in my vsCode Terminal every time the page gets rendered.

I checked the format of the ID and this is an ID generated by mongoDB. 
My POST and also DELETE Method doesn't work and the only ERR I get is the "CastError".

Can someone help me to solve this issue? 

[PR](https://github.com/jonatsuji/capstone-project/pull/10)

detailed ERR message:
![Screenshot 2023-01-18 at 11 28 50](https://user-images.githubusercontent.com/115465230/213149868-84bf8478-bdcf-4ff5-9d4e-72600ca9cb23.png)

logged ID from browser console:
![Screenshot 2023-01-18 at 11 29 34](https://user-images.githubusercontent.com/115465230/213149929-feed3a46-fa8a-4cda-879e-34d160670583.png)

many thanks in advance!

**Comment by ThomasKirst** [comment, marked as answer]
As discussed, the parameter (`_id` instead of `id`) was not passed to the API route.
